src/styles/_card.scss

```diff
@@ -0,0 +1,97 @@
+@import './variables';
+
+.card {
+  display: flex;
+  flex-direction: column;
+  justify-content: space-between;
+  font-family: $font-family-main;
+  box-sizing: border-box;
+  width: 199px;
+  padding: $padding-large $padding-medium $padding-medium;
+  border: 1px solid $border-color;
+  background-color: $secondary-color;
+
+  &__product-image {
+    display: block;
+    width: 160px;
+    height: 134px;
+    top: 33px;
+    left: 20px;
+  }
+
+  &__product-name {
+    padding-top: 40px;
+    font-size: $font-size-medium;
+    font-weight: 500;
+    line-height: 18px;
+    color: $text-color-primary;
+  }
+
+  &__product-code {
+    padding-top: $padding-small;
+    font-size: $font-size-small;
+    font-weight: 400;
+    line-height: 140%;
+    color: $text-color-secondary;
+  }
+
+  &__product-rating {
+    padding-top: $padding-medium;
+    display: flex;
+    flex-direction: row;
+    justify-content: space-between;
+    align-items: center;
+
+    &-reviews {
+      display: flex;
+      font-size: $font-size-small;
+      line-height: 140%;
+      font-weight: 400;
+      color: $text-color-primary;
+      text-align: right;
+    }
+  }
+
+  &__product-price {
+    padding-top: 24px;
+    padding-bottom: $padding-medium;
+    display: flex;
+    flex-direction: row;
+    justify-content: space-between;
+    line-height: 18px;
+
+    &-text {
+      font-size: $font-size-medium;
+      font-weight: 400;
+      color: $text-color-secondary;
+    }
+
+    &-numbers {
+      font-size: $font-size-large;
+      font-weight: 700;
+      color: $text-color-primary;
+    }
+  }
+
+  &__product-buy-button {
+    display: flex;
+    flex-direction: column;
+    justify-content: center;
+    width: 100%;
+    height: 40px;
+    border-radius: $border-radius;
+    background-color: $primary-color;
+    font-size: 14px;
+    line-height: 16px;
+    font-weight: 700;
+    text-decoration: none;
+    text-align: center;
+    color: $secondary-color;
+
+    &:hover {
+      background-color: $secondary-color;
+      border: 1px solid $primary-color;
+      color: $primary-color;
+    }
```

Comment:
In the hover state of the `__product-buy-button`, you're changing the `background-color`, `border`, and `color`. This will cause a repaint of the button, which may lead to performance issues. It's better to only change properties that do not cause a repaint, such as `transform` or `opacity`.